### PR TITLE
[Enhancemenet] Show automatic redemption route even if insufficient funds are available

### LINKED ISSUE #1299: [Enhancemenet] Show automatic redemption route even if insufficient funds are available
- Show routes for both 1tx automatic redeems and 2tx manual redeems, with the 1tx automatic redeem grayed out with a tooltip or warning if the transferred amount is below the minimum amount. E.g. "Not enough funds on the source chain for automatic redemption" or "X more <token> is needed on the source chain for automatic redemption"
- Cosmos routes: do not show 2tx options as currently
- If a CCTP route is available between the source and the destination, do not show the wormhole token bridge route to avoid liquidity fragmentation.

### PR DESCRIPTION
**Pull request by yuli-ferna**
Issue: https://github.com/wormhole-foundation/wormhole-connect/issues/1299

Now we take into account a new field, that the route is supported, in that context it means that the amount is sufficient to cover the fee, even though the route is enabled.

Result:
image

[testt2.webm](https://github.com/wormhole-foundation/wormhole-connect/assets/35125931/b1ef64c3-bf47-4250-91fa-250344849c94)

### CODE REVIEW

File: wormhole-connect/src/routes/relay/relay.ts [hunk 1]
```diff
@@ -83,7 +83,18 @@ export class RelayRoute extends BridgeRoute implements RelayAbstract {
     const accepted = await isAcceptedToken(tokenId);
 
     if (!accepted) return false;
+    return true;
```
Comment (M-Picco) [inline wormhole-connect/src/routes/relay/relay.ts]: You can remove the `accepted` variable and just do `return isAcceptedToken(tokenId);`

File: wormhole-connect/src/routes/cctpRelay/cctpRelay.ts [hunk 2]
```diff
@@ -188,6 +188,16 @@ export class CCTPRelayRoute extends CCTPManualRoute implements RelayAbstract {
       CHAINS[destChainName]?.contracts.cctpContracts?.wormholeCircleRelayer;
     if (!bothHaveRelayer) return false;
 
+    return true;
```
Comment (M-Picco) [inline wormhole-connect/src/routes/cctpRelay/cctpRelay.ts]: You can just return `bothHaveRelayer`

File: wormhole-connect/src/utils/transferValidation.ts [hunk 3]
```diff
@@ -269,7 +271,7 @@ export const validate = async (
     transferInput.destToken &&
     transferInput.amount &&
     Number.parseFloat(transferInput.amount) >= 0 &&
-    transferInput.availableRoutes !== undefined
+    transferInput.routeStates?.filter((rs) => rs.available) !== undefined
```
Comment (M-Picco) [inline wormhole-connect/src/utils/transferValidation.ts]: If you need to enable the validation only if there's at lease one available route you could use `transferInput.routeStates?.some(rs => rs.available)` and do away with the ternary operator

**Review by M-Picco — COMMENTED** [review]
A naming suggestion: swap `supported` and `available`. The first communicates that the token can be sent through the route while the later that you can do it at this moment of time. Then, an `unsupported` route is for example a token that is not accepted by the relayers and an `unavailable` route is one you can send the token through but not now, for whatever reason (e.g. not enough funds to cover the fee).

What do you think?

**Review by M-Picco — APPROVED** [review]

**Review by M-Picco — APPROVED** [review]

**Review by sebastianscatularo — APPROVED** [review]

**Pull request merged**